escape_roomba/thread_manager.py

```diff
@@ -154,28 +151,30 @@ async def _async_message_update(self, channel_id=None, channel=None,
                 ThreadChannel.relevant_origin_update(
                     emoji=emoji, message=message)):
             async with self._message_exclusive.locker(mi):  # Lock & refresh.
-                t = self._thread_by_origin.get(ci, {}).get(mi)
-                if t is not None:  # Message is an existing thread origin.
-                    if not await t.async_refresh_origin():  # Thread deleted!
+                thread = self._thread_by_origin.get(ci, {}).get(mi)
+                if thread is not None:  # Message is an existing thread origin.
+                    await thread.async_refresh_origin()
+                    if thread.is_deleted:  # Thread deleted!
                         del self._thread_by_origin[ci][mi]
-                        del self._thread_by_channel[t.thread_channel.id]
+                        del self._thread_by_channel[thread.thread_channel.id]
                 else:  # No thread based on this message; maybe make one?
-                    t = await ThreadChannel.async_maybe_create_from_origin(
+                    thread = await ThreadChannel.async_maybe_create_from_origin(
```

Comment:
You sure you can afford all these characters? 🤔

Reply:
😛